src/app/modules/form/form.utils.ts

```diff
@@ -0,0 +1,57 @@
+import { pick } from 'lodash'
+import { Merge } from 'type-fest'
+
+import { IPopulatedForm } from 'src/types'
+
+// Kept in this file instead of form.types.ts so that this can be kept in sync
+// with FORM_PUBLIC_FIELDS more easily.
+type PublicFormValues = Pick<
+  IPopulatedForm,
+  | 'authType'
+  | 'endPage'
+  | 'esrvcId'
+  | 'form_fields'
+  | 'form_logics'
+  | 'hasCaptcha'
+  | 'publicKey'
+  | 'startPage'
+  | 'status'
+  | 'title'
+  | '_id'
+  | 'responseMode'
+>
+
+type PublicForm = Merge<
+  PublicFormValues,
+  { admin: Pick<IPopulatedForm['admin'], 'agency'> }
+>
+
+const FORM_PUBLIC_FIELDS = [
+  'admin',
+  'authType',
+  'endPage',
+  'esrvcId',
+  'form_fields',
+  'form_logics',
+  'hasCaptcha',
+  'publicKey',
+  'startPage',
+  'status',
+  'title',
+  '_id',
+  'responseMode',
+]
+
+/**
+ * Removes all private details such as admin email from given form.
+ * @param form the form to scrub
+ * @returns form with only public details
+ */
+export const removePrivateDetailsFromForm = (
+  form: IPopulatedForm,
+): PublicForm => {
+  return {
+    ...(pick(form, FORM_PUBLIC_FIELDS) as PublicFormValues),
```

Comment:
does it not work without this casting? seems a bit futile to declare the type so explicitly and then only use it in a cast haha

Reply:
problem is with `pick`, haha! It returns a `Partial<...>` (as expected la, since it doesn't know what exists and what doesn't) so it requires a cast